pkg/object/bunny.go

```diff
@@ -0,0 +1,132 @@
+/*
+ * JuiceFS, Copyright 2024 Juicedata, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package object
+
+import (
+	"bytes"
+	"io"
+	"net/url"
+	"time"
+
+	bunnystorage "github.com/l0wl3vel/bunny-storage-go-sdk"
+)
+
+type bunnyClient struct {
+	DefaultObjectStorage
+	client   *bunnystorage.Client
+	endpoint string
+}
+
+// Description of the object storage.
+func (b *bunnyClient) String() string {
+	return b.endpoint
+}
+
+// Limits of the object storage.
+func (b *bunnyClient) Limits() Limits {
+	return Limits{
+		IsSupportMultipartUpload: false,
+		IsSupportUploadPartCopy:  false,
+	}
+}
+
+// Get the data for the given object specified by key.
+func (b *bunnyClient) Get(key string, off int64, limit int64) (io.ReadCloser, error) {
+	body, err := b.client.Download(key)
```

Comment:
Can it do range get? When is it will be closed?

Reply:
I just took a look and the GET endpoint indicates support for `accept-ranges: bytes`. I am going to implement it into my library and remove the range read emulation with the section reader.

Reply:
I implemented the download using ranges.